sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -47,9 +47,10 @@ import org.apache.spark.sql.types._
  * (1, 5)
  * (4, 6)
  *
- * @param catalog  the catalog to use for looking up the schema of INSERT INTO table objects.
+ * @param resolveRelations rule to resolve relations from the catalog. This should generally map to
+ *                         the the ResolveRelations analyzer rule.
  */
-case class ResolveDefaultColumns(catalog: SessionCatalog) extends Rule[LogicalPlan] {
+case class ResolveDefaultColumns(resolveRelations: Rule[LogicalPlan]) extends Rule[LogicalPlan] {
```

Comment:
@dtenedor Let's not pass another rule as a parameter. All we need is the method
```
def resolveRelation(
        u: UnresolvedRelation,
        timeTravelSpec: Option[TimeTravelSpec] = None)
```
within the rule `ResolveRelations`

Reply:
Or we can just put the following code(except the pattern match `case i @ InsertIntoStatement...`) as a new method in ResolveRelations
```
      case i @ InsertIntoStatement(table, _, _, _, _, _) if i.query.resolved =>
        val relation = table match {
          case u: UnresolvedRelation if !u.isStreaming =>
            resolveRelation(u).getOrElse(u)
          case other => other
        }

        // Inserting into a file-based temporary view is allowed.
        // (e.g., spark.read.parquet("path").createOrReplaceTempView("t").
        // Thus, we need to look at the raw plan if `relation` is a temporary view.
        unwrapRelationPlan(relation) match {
          case v: View =>
            throw QueryCompilationErrors.insertIntoViewNotAllowedError(v.desc.identifier, table)
          case other => i.copy(table = other)
        }
```

Reply:
Sure, I added a new method in `ResolveRelations` for this purpose.